integration-snippets/controller-init.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# This script checks if upgrade is needed and returns 0 if no upgrade
+# is needed. If upgrade is needed then check if it is in progress and
+# wait until it is finished before checking again.
+#
+# Requirements:
+#  - calico-upgrade is available in the PATH
+
+
+calico-upgrade $UPGRADE_OPTIONS needed $UPGRADE_ARGS
+needed=$?
+while [ $needed -eq 0 ]; do
+    sleep 5
+    calico-upgrade $UPGRADE_OPTIONS needed $UPGRADE_ARGS
+    needed=$?
+done
+
+if [ $needed -eq 1 ]; then
+	echo "No data migration is needed. Continuing on."
```

Comment:
The comment at the top says that if `$needed` is `0`, then no upgrade is needed. I think it conflicts with the handling in this check?

Reply:
I think maybe I worded that poorly.  Maybe it would be clearer phrased this way. `This script checks if upgrade is needed and if no upgrade is needed then returns 0.`
WDYT?

Reply:
Oh, are the data migration and the upgrade considered separate? I was treating them as different parts of the same process in my head, which was why if `$needed` was `1` (meaning that an upgrade was necessary), then sending a message saying you don't need data migration seemed to conflict. If those are 2 separate things, then I guess the message and handling makes more sense.

Reply:
Oh yea, I think I got the variable `$needed` and the return value mixed up.